src/main/java/org/truffleruby/RubyContext.java

```diff
@@ -178,8 +184,28 @@ public RubyContext(RubyLanguage language, TruffleLanguage.Env env) {
         // Create objects that need core classes
 
         Launcher.printTruffleTimeMetric("before-create-native-platform");
-        nativePlatform = NativePlatformFactory.createPlatform(this);
-        featureLoader.initialize(nativePlatform);
+
+        truffleNFIPlatform = options.NATIVE_INTERRUPT ? new TruffleNFIPlatform(this) : null;
+
+        DefaultRubiniusConfiguration.load(rubiniusConfiguration, this);
+
+        switch (Platform.OS) {
+            case LINUX:
+                LinuxRubiniusConfiguration.load(rubiniusConfiguration, this);
+                break;
+            case DARWIN:
+                DarwinRubiniusConfiguration.load(rubiniusConfiguration, this);
+                break;
+            case SOLARIS:
+                SolarisSparcV9RubiniusConfiguration.load(rubiniusConfiguration, this);
+                break;
+            default:
+                Log.LOGGER.severe("no Rubinius configuration for this platform");
+                break;
+        }
+
+        featureLoader.initialize(this);
```

Comment:
The idea here was to pass the dependencies to `featureLoader.initialize`, specifically NFI and the Rubinius config, which were packaged as the NativePlatform.
This makes it really clear what needs to be done in which order, while you would need to dig into the code and check every usage of the passed context otherwise.
Could you pass explicitly NFI and the Rubinius config here, and not the context?

Reply:
Similarly, `TruffleNFIPlatform#resolveType()` should be pass the RubiniusConfiguration, not the full RubyContext.

Reply:
Fixed.